test_runner/regress/test_sharding.py

```diff
@@ -225,6 +225,12 @@ def test_sharding_split_compaction(neon_env_builder: NeonEnvBuilder, failpoint:
     workload.validate()
     workload.stop()
 
+    # Do a full image layer generation before splitting, so that when we compact after splitting
+    # we should only see sizes decrease (from post-split drops/rewrites), not increase (from image layer generation)
+    env.get_tenant_pageserver(tenant_id).http_client().timeline_compact(
+        tenant_id, timeline_id, force_image_layer_creation=True
```

Comment:
This needs to await uploads, I added an arg wait uploads.

Reply:
We discussed this in the call just now, you had the idea that perhaps as part of shard splitting the parent should always be awaited to complete remote ops, or flushed, I forgot which already.

Reply:
I didn't know about `wait_until_uploaded` -- add that here.